oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,305 @@
+        "Did not get expected result code from TIFFReadEncodedTile(): ",
+        status_or_long.value(), " instead of ", sz));
+  }
+
+  unsigned pixel_status = 1;
+  if (status = CheckCluster(0, buffer_, kCluster0); !status.ok()) {
+    LOG(ERROR) << "CheckCluster failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  if (status = CheckCluster(64, buffer_, kCluster64); !status.ok()) {
+    LOG(ERROR) << "CheckCluster failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  if (status = CheckCluster(128, buffer_, kCluster128); !status.ok()) {
+    LOG(ERROR) << "CheckCluster failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  if (!pixel_status) {
+    return absl::InternalError("unexpected pixel_status value");
+  }
+
+  SAPI_ASSIGN_OR_RETURN(
+      status_or_int,
+      api.TIFFSetFieldU1(&tif, TIFFTAG_JPEGCOLORMODE, JPEGCOLORMODE_RGB));
+  if (!status_or_int.value()) {
+    return absl::InternalError("TIFFSetFieldU1 not available");
+  }
+
+  SAPI_ASSIGN_OR_RETURN(status_or_long, api.TIFFTileSize(&tif));
+  if (status_or_long.value() != 128 * 128 * 3) {
+    return absl::InternalError(
+        absl::StrCat("tiles are ", status_or_long.value(), " bytes"));
+  }
+  sz = status_or_long.value();
+
+  sapi::v::Array<uint8_t> buffer2_(sz);
+
+  SAPI_ASSIGN_OR_RETURN(
+      status_or_long, api.TIFFReadEncodedTile(&tif, 9, buffer2_.PtrBoth(), sz));
+  if (status_or_long.value() != sz) {
+    return absl::InternalError(absl::StrCat(
+        "Did not get expected result code from TIFFReadEncodedTile(): ",
+        status_or_long.value(), " instead of ", sz));
+  }
+
+  pixel_status = 1;
+  if (status = CheckRgbPixel(0, 15, 18, 0, 0, 18, 41, buffer2_); !status.ok()) {
+    LOG(ERROR) << "CheckRgbPixel failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  if (status = CheckRgbPixel(64, 0, 0, 0, 0, 0, 2, buffer2_); !status.ok()) {
+    LOG(ERROR) << "CheckRgbPixel failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  if (status = CheckRgbPixel(512, 5, 6, 34, 36, 182, 196, buffer2_);
+      !status.ok()) {
+    LOG(ERROR) << "CheckRgbPixel failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  SAPI_RETURN_IF_ERROR(api.TIFFClose(&tif));
+
+  SAPI_ASSIGN_OR_RETURN(
+      status_or_tif, api.TIFFOpen(srcfile_var.PtrBefore(), r_var.PtrBefore()));
+
+  sapi::v::RemotePtr tif2(status_or_tif.value());
+  if (!tif2.GetValue()) {  // tif is NULL
+    return absl::InternalError(absl::StrCat("Could not reopen ", srcfile));
+  }
+
+  sapi::v::Array<uint32> rgba_buffer_(128 * 128);
+
+  SAPI_ASSIGN_OR_RETURN(
+      status_or_int,
+      api.TIFFReadRGBATile(&tif2, 1 * 128, 2 * 128, rgba_buffer_.PtrBoth()));
+  if (!status_or_int.value()) {
+    return absl::InternalError("TIFFReadRGBATile() returned failure code");
+  }
+
+  if (status = CheckRgbaPixel(0, 15, 18, 0, 0, 18, 41, 255, 255, rgba_buffer_);
```

Comment:
Add a comment what this is supposed to do. "Checking specific pixels from the test data"?